### ISSUE TITLE
[MDEPLOY-37] On Windows, Maven requires local distribution repository URL be specified as invalid file URL with Windows path.

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-deploy-plugin#270]
**[Steven Coco](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=steevcoco)** opened **[MDEPLOY-37](https://issues.apache.org/jira/browse/MDEPLOY-37?redirect=false)** and commented

On Windows, a local distribution repository URL must be specified as an invalid file URL with a Windows file path syntax.  If it instead is given in valid file URL syntax then Maven will treat it as a Windows path name and actually create new directories matching the encoded URL's literal textual value.

Basically, the distribution management element is specified like so:

```xml
<distributionManagement>
    <repository>
        <id>local-release-repository</id>
        <name>Local Release Repository</name>
        
        <!--
          - Notice that this is a Windows path: it uses the backslash:
          - and also the space is not URL encoded.
          - Change this to a valid file URL:
          -      file:///C:/Local%20Repository
          - and Maven will then create a directory named literally:
          -      "Local%20Repository"
          - at the root of the C drive,and deploy there.
         -->
        <url>file:///C:\Local Repository</url>
    </repository>
</distributionManagement>
```

A test case is attached that shows the behavior.  Execute the default goal on a Windows machine; and then also read the comment inside the POM to see more.

---

**Affects:** 2.0

**Attachments:**
- [deploy-path-bug.zip](https://issues.apache.org/jira/secure/attachment/12715259/deploy-path-bug.zip) (_2.62 kB_)

**Issue Links:**
- [MNG-4087](https://issues.apache.org/jira/browse/MNG-4087) Percent encoded characters in file URLs are not decoded upon deployment
 (_**"duplicates"**_)

3 votes, 1 watchers

**Comment by jira-importer** [comment]
**[Steven Coco](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=steevcoco)** commented

I just noticed that this also happens on Linux.  It's the same problem; and if you specify a URL with an encoded space in it, Maven will create a new directory with "%20" in the name.

And this bug's posting is a bit munged by the automatic formatting, but all the information is essentially visible; ignore the links and the bullets in that auto-generated markup.

**Comment by jira-importer** [comment]
**[Benjamin Bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann)** commented

This is actually bug in the Maven core and is fixed in the upcoming 2.1.0 release.
